### ISSUE TITLE
-XDELETE wouldn't delete all the indexes specified

### ISSUE THREAD

**Issue by bashok001** [elastic/elasticsearch#15487]
I have 550 indexes, all in the format of `ashok_dev*`, that I wanted to flush out of my system. I issued this:
```
$ curl -XDELETE http://localhost:9200/ashok_dev_*
{"acknowledged":true}
```
But I still have 459 indexes. I had to run the `-XDELETE` command about 14 times to delete all the indexes.

**Comment by bashok001** [comment]
The number of indexes deleted in each run were varying too.

**Comment by javanna** [comment]
Hi @bashok001 I think you are hitting #7295 which was recently fixed. The fix will be included in the upcoming 2.2 version.

**Comment by bashok001** [comment]
@javanna That's right. Thank you for pointing it out.
